File: api-generator/ag-system/src/main/java/cn/youngkbt/ag/system/model/po/Report.java
package cn.youngkbt.ag.system.model.po;

import cn.youngkbt.ag.system.model.vo.ReportVO;
import cn.youngkbt.mp.annotation.FieldValueFill;
import cn.youngkbt.mp.annotation.ValueStrategy;
import cn.youngkbt.mp.base.BaseDO;
import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import io.github.linpeilie.annotations.AutoMapper;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Kele-Bingtang
 * @date 2024-03-23 01:03:52
 * @note 报表
*/
@EqualsAndHashCode(callSuper = true)
@TableName(value = "t_report")
@Data
@AutoMapper(target = ReportVO.class, reverseConvertGenerate = false)
public class Report extends BaseDO {
    /**
     * 报表 ID
     */
    @TableField(fill = FieldFill.INSERT)
    @FieldValueFill(ValueStrategy.SNOWFLAKE)
    private String reportId;

    /**
     * 报表名称
     */
    private String reportTitle;

    /**
     * 报名描述
     */
    private String description;

    /**
     * 是否允许新增（0 不允许，1 允许）
     */
    private Integer allowAdd;

    /**
     * 是否允许编辑（0 不允许，1 允许）
     */
    private Integer allowEdit;

    /**
     * 是否允许删除（0 不允许，1 允许）
     */
    private Integer allowRemove;

    /**
     * 是否允许查询（0 不允许，1 允许）
     */
    private Integer allowFilter;

    /**
     * 是否允许导出（0 不允许，1 允许）
     */
    private Integer allowExport;

    /**
     * 报表是否出现行数（0 不允许，1 允许）
     */
    private Integer allowRow;

    /**
     * 弹出框宽度
     */
    private String dialogWidth;

    /**
     * 一页显示多少条数据
     */
    private Integer pageSize;

    /**
     * 是否开启图标，0 不开启，1 饼图，2 折线图
     */
    private Integer chartType;

    /**
     * 目录 ID
     */
    private String categoryId;

    /**
     * 服务 ID
     */
    private String serviceId;

    /**
     * 项目 ID
     */
    private String projectId;
    
    /**
     * 团队 ID
     */
    private String teamId;
}

File: work-web/api-generator/src/api/report.ts
import http from "@/config/request";
export declare namespace Report {
  interface ReportInfo {
    id: number;
    reportId: string; // 报表 ID
    reportTitle: string; // 报表名称
    description: string; // 报名描述
    allowAdd: number; // 是否允许新增（0 不允许，1 允许）
    allowEdit: number; // 是否允许编辑（0 不允许，1 允许）
    allowRemove: number; // 是否允许删除（0 不允许，1 允许）
    allowFilter: number; // 是否允许查询（0 不允许，1 允许）
    allowExport: number; // 是否允许导出（0 不允许，1 允许）
    allowRow: number; // 报表是否出现行数（0 不允许，1 允许）
    dialogWidth: string; // 弹出框宽度
    pageSize: number; // 一页显示多少条数据
    chartType: number; // 是否开启图标，0 不开启，1 饼图，2 折线图
    serviceId: string; // 服务 ID
    categoryId: string; // 分类 ID
    projectId: string; // 项目 ID
    teamId: string; // 团队 ID
  }

  interface ReportConfig {
    report: Report.ReportInfo;
    proTableColumnsList: Record<string, any>[];
    proFormSchemaList: Record<string, any>[];
  }
}

const baseUri = "/report";
export const listReportConfig = (serviceId: string) => {
  return http.get<http.Response<Report.ReportConfig>>(`${baseUri}/listReportConfig/${serviceId}`);
};
